### PR TITLE
added composer 2 support

### PR DESCRIPTION
**Pull request by niepi**
- installs both versions of composer
- upgrades composer 1 to 1.10.17
- installs composer 2 at 2.0.7
- environment variable `COMPOSER_VERSION` decides which version is used, defaults to version 1

### CODE REVIEW

File: 7.4/Dockerfile [hunk 1]
```diff
@@ -84,7 +89,8 @@ COPY start-cron /usr/sbin
 RUN chmod +x /usr/sbin/start-cron
 # Cron
 
-ENV PHP_DATE_TIMEZONE="" \
+ENV COMPOSER_MAYOR_VERION=1 \
```
Comment (OskarStark) [inline 7.4/Dockerfile:92]:
```suggestion
ENV COMPOSER_VERSION=1 \
```

File: 7.4/Dockerfile-alpine [hunk 2]
```diff
@@ -235,7 +240,8 @@ RUN chmod +x /usr/sbin/start-cron
 
 ADD php.ini /usr/local/etc/php/php.ini
 
-ENV PHP_DATE_TIMEZONE="" \
+ENV COMPOSER_MAYOR_VERION=1 \
```
Comment (OskarStark) [inline 7.4/Dockerfile-alpine:243]:
```suggestion
ENV COMPOSER_VERSION=1 \
```

File: composer.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+if [ -n "$COMPOSER_MAYOR_VERION" ]
```
Comment (OskarStark) [inline composer.sh:2]:
```suggestion
if [ -n "$COMPOSER_VERSION" ]
```

File: composer.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+then
+    /usr/local/bin/composer$COMPOSER_MAYOR_VERION "$@"
```
Comment (OskarStark) [inline composer.sh:4]:
```suggestion
    /usr/local/bin/composer$COMPOSER_VERSION "$@"
```

File: composer.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+else
+    /usr/local/bin/composer1 "$@"
+fi
+
```
Comment (OskarStark) [inline composer.sh:8]:
```suggestion
```

**Review by OskarStark — CHANGES_REQUESTED** [review]

**Review by niepi — COMMENTED** [review]
Apply suggestions from code review

**Comment by llupa** [comment]
When will this be merged?

**Comment by OskarStark** [comment]
Thanks @niepi I am going to merge this.

@llupa can you please report back if everything works fine?

Afterwards please backport this feature to the other branches please. Thanks

**Comment by niepi** [comment]
I will create a new PR to add documentation

**Pull request merged**